lib/train/transports/ssh.rb

```diff
@@ -86,16 +87,30 @@ def connection(state = {}, &block)
       end
     end
 
-    # Returns the ssh config option like user, port from config files
-    # Params options [Hash], option_type [String]
-    # Return String
-    def self.read_options_from_ssh_config(options, option_type)
-      config_options = Net::SSH.configuration_for(options[:host], true)
-      config_options[option_type]
+    def apply_ssh_config(host)
+      if options[:ssh_config] != false && !options[:ssh_config].nil?
+        files = options[:ssh_config] == true ? Net::SSH::Config.default_files : options[:ssh_config]
+        host_cfg = ssh_config_for_host(host, files)
+        host_cfg.each do |key, value|
+          if key == :keys && options[:key_files].nil? && !host_cfg[:keys].nil? && options[:password].nil?
+            options[:key_files] = host_cfg[key]
```

Comment:
I had to re-read this a few times and read the tests before I understood that this was simply remapping the private keys option. A comment to that effect would be helpful.